mono/metadata/reflection.c

```diff
@@ -8845,8 +8853,19 @@ mono_reflection_create_custom_attr_data_args (MonoImage *image, MonoMethod *meth
 				return;
 			}
 
-			obj = (MonoObject *)(type_is_reference (field->type) ? val : mono_value_box (domain, mono_class_from_mono_type (field->type), val));
+			if (type_is_reference (field->type)) {
+				obj = (MonoObject*) val;
+			} else {
+				obj = mono_value_box_checked (domain, mono_class_from_mono_type (field->type), val, error);
+				if (!mono_error_ok (error)) {
+					g_free (val);
+					g_free (name);
+					return;
```

Comment:
As of here, I used the same error handling as https://github.com/mono/mono/pull/2495/files#diff-2a19a46e927fb09f5f3bc8cbef8e2d4cR8890 and https://github.com/mono/mono/pull/2495/files#diff-2a19a46e927fb09f5f3bc8cbef8e2d4cR8849